lib/ui/window.dart

```diff
@@ -914,31 +914,76 @@ class Window {
   /// A callback that is invoked to report the [FrameTiming] of recently
   /// rasterized frames.
   ///
-  /// This can be used to see if the application has missed frames (through
-  /// [FrameTiming.buildDuration] and [FrameTiming.rasterDuration]), or high
-  /// latencies (through [FrameTiming.totalSpan]).
-  ///
-  /// Unlike [Timeline], the timing information here is available in the release
-  /// mode (additional to the profile and the debug mode). Hence this can be
-  /// used to monitor the application's performance in the wild.
-  ///
-  /// {@macro dart.ui.TimingsCallback.list}
-  ///
-  /// If this is null, no additional work will be done. If this is not null,
-  /// Flutter spends less than 0.1ms every 1 second to report the timings
-  /// (measured on iPhone6S). The 0.1ms is about 0.6% of 16ms (frame budget for
-  /// 60fps), or 0.01% CPU usage per second.
+  /// This is deprecated, use [frameTimingStream] instead.
+  @Deprecated('Use frameTimingStream instead.')
   TimingsCallback get onReportTimings => _onReportTimings;
   TimingsCallback _onReportTimings;
   Zone _onReportTimingsZone;
+  @Deprecated('Use frameTimingStream instead.')
   set onReportTimings(TimingsCallback callback) {
+    _internalSetOnReportTimings(callback);
+  }
+
+  void _internalSetOnReportTimings(TimingsCallback callback) {
     if ((callback == null) != (_onReportTimings == null)) {
       _setNeedsReportTimings(callback != null);
     }
     _onReportTimings = callback;
     _onReportTimingsZone = Zone.current;
   }
 
+  /// Mock the calling of [_onReportTimings] for unit tests.
+  void debugReportTimings(List<FrameTiming> timings) {
+    _onReportTimings(timings);
+  }
+
+  /// Check whether the engine has to report timings.
+  ///
+  /// This is for unit tests and debug purposes only.
+  bool get debugNeedsReportTimings => _onReportTimings != null;
+
+  StreamController<FrameTiming> _frameTimingBroadcastController;
+
+  void _onFrameTimingListen() {
+    _internalSetOnReportTimings((List<FrameTiming> timings) {
+      timings.forEach(_frameTimingBroadcastController.add);
+    });
+  }
+
+  // If there's no one listening, set [onReportTimings] back to null so the
+  // engine won't send [FrameTiming] from engine to the framework.
+  void _onFrameTimingCancel() {
+    _internalSetOnReportTimings(null);
+  }
+
+  /// A broadcast stream of the frames' time-related performance metrics.
+  ///
+  /// This can be used to see if the application has missed frames (through
+  /// [FrameTiming.buildDuration] and [FrameTiming.rasterDuration]), or high
+  /// latencies (through [FrameTiming.totalSpan]).
+  ///
+  /// Unlike [Timeline], the timing information here is available in the release
```

Comment:
drop the "the" before "release", "profile" and "debug"?

Reply:
Done.